### PR TITLE
Updated Input with new features

### PR DESCRIPTION
**Pull request by PiyushPawar17**

### CODE REVIEW

**Comment by arorayash** [comment]
Hey, went through the cases, could you please add another case with limiting the resize option for textarea to one axis?

- The overflow has no gradient. It appears that an solid color fill is being used.
- Background fill is missing from the close icon in Search input
- Padding/distance between the icon and the text is uneven - The icon should have a uniform padding on all sides.
- Add a transition when adding border on hover
- Number field shouldn't allow adding alphabetical inputs
- Alignment of the password show/hide icon seems off from the right. It should be 1rem.

File: src/components/Input/utils.ts [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+  }
+}
+
+export const getFontSize = (props: IStyledInput) => {
+  const {
+    customProps: { inputSize },
+    theme: { knitui },
+  } = props
+  const typographySize = inputSizeToTypographySize[inputSize!]
+  return knitui.typography[typographySize].fontSize
```
Comment (prasook-jain) [inline src/components/Input/utils.ts]: please use `getThemeColor`, `getThemeFontSize`, `getThemeLineHeight` from `common/_utils/`

File: src/components/Input/__tests__/Input.test.tsx [hunk 2]
```diff
@@ -153,6 +153,8 @@ describe("Input", () => {
   it("should display disabled icon on hover for disabled input", () => {
     const { container } = render(<Input onChange={() => {}} disabled />)
 
-    expect(container.firstChild).toHaveStyle("cursor: not-allowed")
+    if (container.firstChild) {
+      expect(container.firstChild.firstChild).toHaveStyle("cursor: not-allowed")
+    }
```
Comment (prasook-jain) [inline src/components/Input/__tests__/Input.test.tsx]: Please also cover the test cases of additional functionality & changes.

File: src/components/Input/README.md [hunk 3]
```diff
@@ -26,18 +26,19 @@ const InputWrapper = () => {
...
+| label          | string \| ReactNode                                           | Yes      | None      | content to be shown above the input as a label        |
+| notification   | string \| ReactNode                                           | Yes      | None      | content to be shown below the input as a notification |
+| onChange       | Function                                                      | Yes      | None      | onChange handler                                      |
+| addonAfter     | string \| ReactNode                                           | Yes      | None      | to show after input                                   |
+| addonBefore    | string \| ReactNode                                           | Yes      | None      | to show before input                                  |
+| inputSize      | "default" \| "small" \| "lage"                                | Yes      | "default" | the size of the inputbutton                           |
+| type           | "default" \| "search" \| "number" \| "password" \| "textarea" | Yes      | "default" | Type of input field to be displayed                   |
```
Comment (prasook-jain) [inline src/components/Input/README.md]: please mentions we are not supporting all the html input types value, so it clearly indicate that only support some html input type.

File: src/components/Input/Input.tsx [hunk 4]
```diff
@@ -209,11 +37,59 @@ const RenderInput = React.forwardRef<HTMLElement, IInputProps>((props, ref) => {
...
+          : props.type === "number"
+          ? "number"
+          : "text"
+      }
+      {...commonProps}
+    />
+  )
+})
+
+function renderInputAddons(
```
Comment (prasook-jain) [inline src/components/Input/Input.tsx]: Please use arrow function convention to define functions

Comment (prasook-jain) [inline src/components/Input/Input.tsx]: Better to split this file component wise.

File: src/common/styles/variables.tsx [hunk 5]
```diff
@@ -135,9 +135,14 @@ const generateTheme = palette => {
     inputBorderRadius: "4px",
     inputBorderColor: "transparent",
     inputFocusBorderColor: chromaPalette.Azure80,
-    inputError: "#990000",
-    inputSuccess: "#05b300",
+    inputHoverBorderColor: "hsl(0, 0%, 80%)",
```
Comment (prasook-jain) [inline src/common/styles/variables.tsx]: if this color are present in `chromaPalette` please use it from there.

File: src/components/Input/Input.tsx [hunk 6]
```diff
@@ -271,26 +125,171 @@ function renderInputAddons(
...
+}
+
+const InputRef = props => {
+  const [currentValue, setCurrentValue] = useState(props.value || "")
+  const [isPasswordField, setIsPasswordField] = useState(
+    props.type === "password"
+  )
+  const [isOverflow, setIsOverflow] = useState(false)
+
+  const inputRef = props.ref || useRef<HTMLInputElement>()
```
Comment (prasook-jain) [inline src/components/Input/Input.tsx]: we are doing forwardReferencing here.. please reconsider, ref flow & use-case in this component again.

**Pull request merged**